### PR TITLE
Adding raf to focusElement so Firefox can tab into dialog iframe

### PR DESCRIPTION
**Pull request by LFDanLu**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

**Comment by snowystinger** [comment]
Hmmm there is the potential for the focus to be buried 2 raf's deep, should this take an argument to help it decide that? or separate it? https://github.com/adobe/react-spectrum/blob/a4baa38b2d4771036356d7742e02372faf658d68/packages/%40react-aria/focus/src/FocusScope.tsx#L384

**Comment by LFDanLu** [comment]
@snowystinger aw shoot, yeah got overzealous. It almost feels like any `.focus` should have a wrapping raf since Firefox related issues with focus have happened twice already, gotta think on this

File: packages/@react-aria/focus/src/FocusScope.tsx [hunk 1]
```diff
@@ -298,7 +298,9 @@ function focusElement(element: HTMLElement | null, scroll = false) {
     }
   } else if (element != null) {
     try {
-      element.focus();
+      requestAnimationFrame(() => {
+        element.focus();
+      });
```
Comment (LFDanLu) [inline packages/@react-aria/focus/src/FocusScope.tsx:303]: I tried adding this raf to the `.focus` calls within `focusWithoutScrolling` in an effort to remove it from https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/focus/src/FocusScope.tsx#L382, but it kept breaking a bunch of tests. Settled with this change as a quick fix for release

File: packages/@react-spectrum/dialog/test/Dialog.test.js [hunk 2]
```diff
@@ -69,13 +86,24 @@ describe('Dialog', function () {
     let input2 = getByTestId('input2');
     expect(document.activeElement).toBe(dialog);
 
-    fireEvent.keyDown(document.activeElement, {key: 'Tab'});
+    act(() => {
+      fireEvent.keyDown(document.activeElement, {key: 'Tab'});
```
Comment (snowystinger) [inline packages/@react-spectrum/dialog/test/Dialog.test.js:90]: ah, recently determined fireEvent doesn't need to be in an act, did you run into an issue when it wasn't?

Comment (LFDanLu) [reply packages/@react-spectrum/dialog/test/Dialog.test.js:90]: ah, put it inside the act out of habit, pulling it out seems to work too

**Review by ktabors — APPROVED** [review]

**Review by snowystinger — APPROVED** [review]

**Pull request merged**